### PR TITLE
enhancement: add "ocr_only" strategy for PDFs

### PR DESCRIPTION
**Pull request by MthwRobinson**
### Summary

Adds an "ocr_only" strategy for PDF documents. This is equivalent to the ocr_strategy for images. Also refactors the strategy selection logic into its own module.

### Testing

```python
from unstructured.partition.pdf import partition_pdf

filename = "example-docs/layout-parser-paper-fast.pdf"

with open(filename, "rb") as f:
    elements = partition_pdf(file=f, strategy="ocr_only")

elements = partition_pdf(filename=filename, strategy="ocr_only")
```

### CODE REVIEW

File: docs/source/bricks.rst [hunk 1]
```diff
@@ -363,10 +363,22 @@ if you'd like to run inference locally.
...
+
+The ``strategy`` kwarg controls the method that will be used to process the PDF.
+The available strategies for PDFs are `"hi_res"`, `"ocr_only"`, and `"fast"`.
+The ``"hi_res"`` strategy will identify the layout of the document using ``detectron2``. The advantage of `"hi_res"` is that
+it uses the document layout to gain additional information about document elements. We recommend using this strategy
+if your use case is highly sensitive to correct classifications for document elements. If ``detectron2`` is not available,
+the ``"hi_res"`` strategy will fall back to the ``"ocr_only"`` strategy.
+The ``"ocr_only"`` strategy runs the document through Tesseract for OCR and then runs the raw text through ``partition_text``.
+Currently, ``"hi_res"`` has difficulty ordering elements for documents with multiple columns. If you have a document with
+multiple columns that does not have extractable text, we recoomend using the ``"ocr_only"`` strategy. ``"ocr_only"`` falls
```
Comment (qued) [inline docs/source/bricks.rst:375]:
```suggestion
multiple columns that does not have extractable text, we recommend using the ``"ocr_only"`` strategy. ``"ocr_only"`` falls
```

File: docs/source/bricks.rst [hunk 2]
```diff
@@ -411,6 +423,17 @@ Examples:
...
+
+The ``strategy`` kwarg controls the method that will be used to process the PDF.
+The available strategies for images are `"hi_res"` and ``"ocr_only"``.
+The ``"hi_res"`` strategy will identify the layout of the document using ``detectron2``. The advantage of `"hi_res"` is that it
+uses the document layout to gain additional information about document elements. We recommend using this strategy
+if your use case is highly sensitive to correct classifications for document elements. If ``detectron2`` is not available,
+the ``"hi_res"`` strategy will fall back to the ``"ocr_only"`` strategy.
+The ``"ocr_only"`` strategy runs the document through Tesseract for OCR and then runs the raw text through ``partition_text``.
+Currently, ``"hi_res"`` has difficulty ordering elements for documents with multiple columns. If you have a document with
+multiple columns that does not have extractable text, we recoomend using the ``"ocr_only"`` strategy.
```
Comment (qued) [inline docs/source/bricks.rst:435]:
```suggestion
multiple columns that does not have extractable text, we recommend using the ``"ocr_only"`` strategy.
```

**Review by qued — APPROVED** [review]
A few typos and a comment/question, but overall LGTM!

In the future I think instead of checking individual models/dependencies we should have something like `is_segmentation_available` and `is_ocr_available` as are various components are made more modular.

File: test_unstructured/partition/test_image.py [hunk 3]
```diff
@@ -162,7 +162,7 @@ def test_partition_image(url, api_called, local_called):
         attribute="_partition_via_api",
         new=mock.MagicMock(),
     ), mock.patch.object(pdf, "_partition_pdf_or_image_local", mock.MagicMock()):
-        image.partition_image(filename="fake.pdf", url=url)
+        image.partition_image(filename="example-docs/layout-parser-paper-fast.pdf", url=url)
```
[the remark this answers is not in the document]
Comment (MthwRobinson) [reply test_unstructured/partition/test_image.py]: Yeah you're right, I changed that back. I think I had test failures at some point along the line that resolved themselves by the time I was done implementing.

**Pull request merged**